tools/apiview/emitters/cadl-apiview/src/apiview.ts

```diff
@@ -0,0 +1,739 @@
+import {
+  ArrayExpressionNode,
+  BaseNode,
+  BooleanLiteralNode,
+  DecoratorExpressionNode,
+  EnumMemberNode,
+  EnumSpreadMemberNode,
+  EnumStatementNode,
+  IdentifierNode,
+  InterfaceStatementNode,
+  IntersectionExpressionNode,
+  MemberExpressionNode,
+  ModelExpressionNode,
+  ModelPropertyNode,
+  ModelSpreadPropertyNode,
+  ModelStatementNode,
+  Namespace,
+  navigateProgram,
+  NumericLiteralNode,
+  OperationSignatureDeclarationNode,
+  OperationSignatureReferenceNode,
+  OperationStatementNode,
+  Program,
+  StringLiteralNode,
+  SyntaxKind,
+  TupleExpressionNode,
+  TypeReferenceNode,
+  UnionExpressionNode,
+} from "@cadl-lang/compiler";
+import { ApiViewDiagnostic, ApiViewDiagnosticLevel } from "./diagnostic.js";
+import { getFullyQualifiedNamespace } from "./emitter.js";
+import { ApiViewNavigation } from "./navigation.js";
+import { generateId, NamespaceModel } from "./namespace-model.js";
+import { LIB_VERSION } from "./version.js";
+
+export const enum ApiViewTokenKind {
+  Text = 0,
+  Newline = 1,
+  Whitespace = 2,
+  Punctuation = 3,
+  Keyword = 4,
+  LineIdMarker = 5, // use this if there are no visible tokens with ID on the line but you still want to be able to leave a comment for it
+  TypeName = 6,
+  MemberName = 7,
+  StringLiteral = 8,
+  Literal = 9,
+  Comment = 10,
+  DocumentRangeStart = 11,
+  DocumentRangeEnd = 12,
+  DeprecatedRangeStart = 13,
+  DeprecatedRangeEnd = 14,
+  SkipDiffRangeStart = 15,
+  SkipDiffRangeEnd = 16,
+}
+
+export interface ApiViewToken {
+  Kind: ApiViewTokenKind;
```

Comment:
typescript convention is to use `camelCase` for properties

Reply:
Yep, in this case, this is the casing needed by APIView. I don't know how to control the JSON serialization to change the casing of the keys (found how to do it with values) so I just kept the casing to match.

Reply:
I see, then yeah that make sense, there isn't an good way to do that without loading a full model + serialization library(Or just converting at the last minute). So should be good like that.